### ISSUE TITLE
Dynamic Add Step

### ISSUE THREAD

**Issue by ricardoborges** [troch/angular-multi-step-form#53]
Is it possible to add new steps on the fly? thanks

**Comment by rdzidziguri** [comment]
What do u mean adding it on the fly?

the way u can achieve this is to implement next step and decide what next step is going to be. u need to load all steps as this is very simple operation just an array of steps and steps are simple objects u can extend them in any way u want 

example of step

``` json
{
    "title": "Step title here",
    "caption": "caption which is shown on wizard",
    "tag": [ "TAG1", "TAG2" ],
    "order": 8,
    "isolatedScope": true,
    "controller": "yourCtrlName",
    "controllerAs": "vm",
    "templateUrl": "/path/to/your/template",
    "hasForm": true
  }
```

**Comment by troch** [comment]
You can push steps to the steps array you pass to the directive. However this is not recommended because the library tracks the active step by index.

"hat you can do is have all your steps in your steps array, and then decide which ones are visible to the user.

**Comment by troch** [comment]
I've released version 1.3.0 which adds support for a custom controller at the main directive level. That should help for this kind of stuff.

**Comment by ricardoborges** [comment]
Roger that! Thank you guys, I'll test the ideas you gave me
